ci/github/install.sh

```diff
@@ -43,7 +43,12 @@ if [[ "$RUNNER_OS" == "macOS" ]] && [[ "${B2_COMPILER:-}" =~ "clang-" ]] && ! co
         B2_COMPILER=clang
     else
         # When the default clang doesn't match the requested version try using the brew installed one
-        if ! B2_COMPILER=$(brew --prefix llvm@$clang_version)/bin/clang; then
+        if brew_clang_prfix=$(brew --prefix llvm@$clang_version); then
+          B2_COMPILER="$brew_clang_prfix/bin/clang"
+          # Workaround for clang linking system libc++ but using internal headers
+          # See https://github.com/Homebrew/homebrew-core/issues/169820
+          mv "$brew_clang_prfix/include/c++/v1" "$brew_clang_prfix/include/c++/v1.bak"
```

Comment:
How does this contrast with the other suggested fix:

"try adding the flag -L"$(brew --prefix llvm)"/lib/c++ "  

Would the other method cause tests to use the brew installed libraries?   That is the targeted test case.

Reply:
The missing stdlib symbols were caused by using clang for linking c++ files.

The new approach works: https://github.com/boostorg/boost-ci/actions/runs/17242657870/job/48924104883